.github/workflows/model_ci.yml

```diff
@@ -1,44 +1,241 @@
 name: Model CI
-on: [push]
+on:
+  push:
+    branches:
+    - 'master'
+  pull_request:
+    types: [opened, synchronize]
+
+env:
+  DOMAIN: 'carbon.rasa-bots.com'
 
 jobs:
   build-model:
     name: Build, test, and upload model
     runs-on: ubuntu-latest
+
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
+
     - name: Set up Python 3.7
       uses: actions/setup-python@v1
       with:
         python-version: 3.7
+
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
         pip install -r requirements.txt
-    #- name: Check stories are consistent
-    #  run: |
-    #    rasa data validate stories --max-history 5 --fail-on-warning
+
+    - name: Check stories are consistent
+      run: |
+        rasa data validate stories --max-history 5 --fail-on-warning
+
     - name: Train model
       run: |
         rasa train
-    # - name: Run Through Test Stories
-    #   run: |
-    #     rasa test --stories test_stories/stories.md --fail-on-prediction-errors
+
+    - name: Run Through Test Stories
+      run: |
+        rasa test core --stories test_stories/stories.md --fail-on-prediction-errors
+
     - name: Cross-validate NLU model
       run: |
         rasa test nlu -f 5 --cross-validation
         python format_results.py
+
+    - name: Upload Cross Validation Results
+      uses: actions/upload-artifact@master
+      with:
+        name: cross-validation-result
+        path: results.md
+
+    - name: Upload model
+      if: github.event_name == 'push' && (startsWith(github.event.ref, 'refs/tags') || github.ref == 'refs/heads/master')
+      uses: actions/upload-artifact@master
+      with:
+        name: model
+        path: models
+
+  share_cross_validation_result:
+    # Separate job to avoid expired GitHub token
+    name: Share cross validation results
+    runs-on: ubuntu-latest
+    needs: [build-model]
+
+    steps:
+    - name: Download Cross Validation Results
+      uses: actions/download-artifact@master
+      with:
+        name: cross-validation-result
+
     - name: post cross-val results to PR
       uses: amn41/comment-on-pr@comment-file-contents
       continue-on-error: true
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       with:
-        msg: results.md
-    - name: Upload model
-      if: github.ref == 'refs/heads/master'
-      uses: actions/upload-artifact@master
+        msg: cross-validation-result/results.md
+
+  build-images:
+    name: Build and Push Images
+    runs-on: ubuntu-latest
+    env:
+      IS_MASTER_BRANCH: ${{ github.event_name == 'push' && (startsWith(github.event.ref, 'refs/tags') || github.ref == 'refs/heads/master') }}
```

Comment:
the name of the env variable is not really correct — it will also be true for tagged non-master branches